src/GitStatusCache/src/Git.cpp

```diff
@@ -270,6 +271,20 @@ bool Git::GetRefStatus(Git::Status& status, UniqueGitRepository& repository)
 		Log("Git.GetRefStatus.NoUpstream", Severity::Spam)
 			<< R"(Branch does not have a remote tracking reference. { "repositoryPath": ")" << status.RepositoryPath
 			<< R"(", "localBranch": ")" << status.Branch << R"(" })";
+
+		auto upstreamName = git_buf{ 0 };
+		result = git_branch_upstream_remote(

```

Comment:
If I'm [reading correctly](https://github.com/libgit2/libgit2/blob/cb76eed5ce41c64607c4f99af947c79c493e8b7e/src/branch.c#L399-L422), this only checks `branch.{name}.remote`. Git doesn't consider an upstream to be valid without `branch.{name}.merge` also being set.

![image](https://cloud.githubusercontent.com/assets/133987/21970659/2f266c20-db6e-11e6-891b-04a70f49aed6.png)